src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -767,14 +767,23 @@ BoundNode bindSpreadElement(BoundCollectionExpressionSpreadElement element, Type
             }
         }
 
-        internal static BoundExpression GetUnderlyingCollectionExpressionElement(BoundExpression element)
+        /// <summary>
+        /// If the element is from a collection type where elements are added with collection initializers,
```

Comment:
>If the element is from a collection type where elements are added with collection initializers, [](http://example.com/codeflow?start=12&length=95)

As someone without very deep involvement into the "Collection Expressions" feature development, I find terminology used in this doc comment very confusing. For example, I have no idea what "elements are added with collection initializers" really means. There are no collection initializers (which is a specific syntactic construct) in the added test scenarios. The same goes for "if the element is not a collection initializer node". It is quite possible I am missing something from the feature specification. Could you provide a link to the relevant section then?

Reply:
The *collection initializers* here are meant to refer to the bound nodes - `BoundCollectionElementInitializer` and `BoundDynamicCollectionElementInitializer` - used in the bound tree to capture the `Add` method invocation for each element, for use in lowering. These bound nodes are an implementation detail only, and *collection initializers* are not part of the feature specification for collection expressions. I agree though that it's confusing to use the term here.